semantic-codeql/src/Language/CodeQL.hs

```diff
@@ -1,9 +1,12 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# OPTIONS_GHC -freduction-depth=0 #-}
```

Comment:
The datatypes here are sufficiently nested and gnarly that GHC has issues proving that typechecking will terminate. We already had this fix deployed for TSX/TypeScript, so I don’t consider it particularly evil.